### PR TITLE
Add RouteActionCallableRector

### PR DESCRIPTION
**Pull request by zingimmick**
Use PHP callable syntax instead of string syntax for controller route declarations, which provides better support for jumping to the controller class in many IDEs.

```diff
use Illuminate\Support\Facades\Route;

-Route::get('/users', 'App\Http\Controllers\UserController@index');
+Route::get('/users', [\App\Http\Controllers\UserController::class, 'index']);
```

### CODE REVIEW

**Comment by zingimmick** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: src/Rector/StaticCall/RouteActionCallableRector.php [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        }
+
+        $arg = $node->args[$position];
+
+        $action = $this->valueResolver->getValue($arg->value);
+        if (! $this->isActionString($action)) {
+            return null;
+        }
+
+        [$controller, $method] = explode('@', $action);
```
Comment (samsonasik) [inline src/Rector/StaticCall/RouteActionCallableRector.php:101]: while technically never met, there should be a check for multiple `@`, eg: `Foo@bar@baz`, .so only applicable when string only have 1 `@` in the middle.

Comment (zingimmick) [reply src/Rector/StaticCall/RouteActionCallableRector.php]: Thanks for reminding, added a test fixture for that.

File: src/Rector/StaticCall/RouteActionCallableRector.php [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+        if ($node instanceof MethodCall && $this->nodeTypeResolver->isObjectTypes(
+            $node->var,
+            [new ObjectType('Illuminate\Routing\Router'), new ObjectType('Illuminate\Routing\RouteRegistrar')]
+        )) {
+            return true;
+        }
+
+        return $node instanceof StaticCall && $this->isName($node->class, 'Illuminate\Support\Facades\Route');
+    }
```
Comment (TomasVotruba) [inline src/Rector/StaticCall/RouteActionCallableRector.php:152]: This looks like complex re-usable logic. Could you extact is to own service?

Comment (zingimmick) [reply src/Rector/StaticCall/RouteActionCallableRector.php]: Extract it to `RouterRegisterNodeAnalyzer`.

**Review by samsonasik — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**